src/vmm_mad/remotes/kvm/migrate

```diff
@@ -122,29 +119,47 @@ else
         exit 1
     fi
 
-    for DISK in $DISKS; do
-        read -r SIZE FORMAT <<<"$(get_size_and_format_of_disk_img "$DISK" "$QEMU_IMG_PARAM")"
+    ssh_monitor_and_log "$DEST_HOST" "mkdir -p '$VM_DIR'" \
+        "Failed to make remote directory $VM_DIR image"
+
+    MIGRATE_DISKS=""
+
+    for DISK_STR in $DISKS; do
+        DISK_DEV=${DISK_STR/,*/}
+        DISK_PATH=${DISK_STR/*,/}
+
+        read -r SIZE FORMAT <<<"$(get_size_and_format_of_disk_img "$DISK_PATH" "$QEMU_IMG_PARAM")"
 
         if [ "$FORMAT" = "raw" ]; then
-            if ! is_readonly $DEPLOY_ID $DISK; then
-                RAW_DISKS+=" $DISK"
+            if ! is_readonly $DEPLOY_ID $DISK_PATH; then
+                RAW_DISKS+=" $DISK_PATH"
+                MIGRATE_DISKS+="${DISK_DEV},"
             fi
+
             # do initial rsync
-            exec_and_log "rsync $DISK $DEST_HOST:$DISK" \
-                "Failed to rsync disk $DISK to $DEST_HOST:$DISK"
+            multiline_exec_and_log "tar -cSf - $DISK_PATH | ssh $DEST_HOST 'tar xSf - -C / '" \
+                "Failed to rsync disk $DISK_PATH to $DEST_HOST:$DISK_PATH"
+
+        elif [ "$FORMAT" = "qcow2" ] && [ ! -L "$DISK_PATH" ]; then
```

Comment:
$DISK_PATH is already tested in get_size_and_format_of_disk_img() line 45
so for the symlinked disks we will have FORMAT=network-disk and this test will be skipped on the first part so the following should be enough (IMO)?

elif [ "$FORMAT" = "qcow2" ]; then